assets/components/button/_sharedButton.jsx

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+// ----- Imports ----- //
+
+import React, { type Node } from 'react';
+
+import SvgArrowRightStraight from 'components/svgs/arrowRightStraight';
+import { classNameWithModifiers } from 'helpers/utilities';
+
+import './button.scss';
+
+// ----- PropTypes ----- //
+
+type Appearance = 'primary' | 'green' | 'greenHollow' | 'greyHollow';
+type IconSide = 'left' | 'right';
+
+type SharedButtonPropTypes = {|
+  children: Node,
+  icon?: Node,
+  appearance: Appearance,
+  iconSide: IconSide,
+  modifierClasses: string[],
+|};
+
+type PropTypes = {
+  element: 'a' | 'button' | 'div',
+  ...SharedButtonPropTypes,
+};
+
+// ----- Render ----- //
+
+const SharedButton = ({
+  element: HtmlElement, appearance, iconSide, modifierClasses, children, icon, ...otherProps
+}: PropTypes) => (
+  <HtmlElement
+    className={classNameWithModifiers('component-button', [
+        appearance,
+        `icon-${iconSide}`,
```

Comment:
and again mirroring https://interactive.guim.co.uk/embed/article-embeds/HiZef/embed.html, what about `hasicon-left` and `hasicon-right`? if that page is still likely to form the basis of some internal standards, it might be a good model on which to base our naming conventions